overlord/configstate/configcore/kernel.go

```diff
@@ -60,15 +60,14 @@ func changedKernelConfigs(c RunTransaction) []string {
 	return changed
 }
 
-func validateCmdlineParamsAreAllowed(st *state.State, devCtx snapstate.DeviceContext, params []string) error {
-	// gd, err := devicestate.CurrentGadgetInfo(st, devCtx)
-	// if err != nil {
-	// 	return err
-	// }
-	// logger.Debugf("gadget data read from %s", gd.RootDir)
-	// TODO use gadgetdata to check against allowed values
+func validateCmdlineParamsAreAllowed(st *state.State, devCtx snapstate.DeviceContext, cmdline string) error {
+	gd, err := devicestate.CurrentGadgetInfo(st, devCtx)
+	if err != nil {
+		return err
+	}
+	logger.Debugf("gadget data read from %s", gd.RootDir)
 
-	return nil
+	return gadget.CheckCmdlineAllowed(cmdline, gd.Info.KernelCmdline.Allow)
```

Comment:
as the gadget can change while the options remain we also need to filter things in devicestate handlers I think

Reply:
the spec discusses this this way:

> cmdline-append is always applied and filtered by the gadget.yaml.

so filtering is the main thing. I agree that it makes sense to error here, but the general defense it the filtering. So in the case of the new gadget/gadget refresh what should happen is filtering out of not allowed anymore argument, and yes the arguments filtered out should probably be logged both on the task and the system log

Reply:
Done now